website_portal_medical_insurance_us/security/medical_insurance_company.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+    Copyright (C) 2016-TODAY LasLabs [https://laslabs.com]
+    @license LGPL-3
+-->
+
+<odoo>
+
+    <record model="ir.rule" id="medical_insurance_company_rule_portal">
+        <field name="name">medical.insurance.company.rule.portal</field>
+        <field name="model_id" ref="medical_insurance.model_medical_insurance_company"/>
+        <field name="groups" eval="[(4, ref('base.group_portal'))]"/>
+        <field name="domain_force">['|',
+            ('create_uid', '=', user.id),
+            ('is_verified', '=', True),
```

Comment:
@lasley I'm having some issues with this security rule, causing Travis to fail.
https://travis-ci.org/LasLabs/vertical-medical/jobs/274413214#L2432

The code generating the plans for the view is [here](https://github.com/LasLabs/vertical-medical/blob/6081c9d8ad1d8cebc4b303341ffbe13b98081cc3/website_portal_medical_insurance_us/models/medical_insurance_plan.py#L26)

Reply:
Seems like create_uid is causing the problem.

Reply:
Why `.sudo()` [here](https://github.com/LasLabs/vertical-medical/blob/6081c9d8ad1d8cebc4b303341ffbe13b98081cc3/website_portal_medical_insurance_us/controllers/main.py#L67) and [here](https://github.com/LasLabs/vertical-medical/blob/6081c9d8ad1d8cebc4b303341ffbe13b98081cc3/website_portal_medical_insurance_us/controllers/main.py#L88). 

It's going to be with the sudo, although usually the issue is on the create. I don't see the create though, so I assume it's happening in `website_form`, which gives good permissions.

Reply:
hmm. Well that was another issue. There were security errors when accessing patient, on product.uom.categ in the wizard on those exact lines. I noticed you used sudo in other parts of the file so figured that might work.

Could just be moving the poop train further down the line so I guess I'll re-look at them.

Reply:
Did you create the test company with the same user you are testing with?

Reply:
Or mark it as verified?

Reply:
Alright so I fixed the issue with the product.uom.categ error (at least locally).

Interestingly, after disabling all the security rules other than the access rules, the issue for accessing the medical plans still persists.

create a new plan as portal user then navigate back to the plans view. Following error happening. The interesting thing is that you clearly have a portal access rule for medical.insurance.company.

Runbot and travis may tell another story once done but this is what I'm getting locally.

```
Traceback (most recent call last):
  File "/opt/odoo/addons/website/models/ir_http.py", line 265, in _handle_exception
    response = super(Http, cls)._handle_exception(exception)
  File "/opt/odoo/odoo/addons/base/ir/ir_http.py", line 169, in _handle_exception
    return request._handle_exception(exception)
  File "/opt/odoo/odoo/http.py", line 766, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/addons/base/ir/ir_http.py", line 195, in _dispatch
    result = request.dispatch()
  File "/opt/odoo/odoo/http.py", line 825, in dispatch
    r = self._call_function(**self.params)
  File "/opt/odoo/odoo/http.py", line 331, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo/service/model.py", line 119, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo/http.py", line 327, in checked_call
    result.flatten()
  File "/opt/odoo/odoo/http.py", line 1257, in flatten
    self.response.append(self.render())
  File "/opt/odoo/odoo/http.py", line 1250, in render
    return env["ir.ui.view"].render_template(self.template, self.qcontext)
  File "/opt/odoo/odoo/addons/base/ir/ir_ui_view.py", line 1029, in render_template
    return self.browse(self.get_view_id(template)).render(values, engine)
  File "/opt/odoo/addons/website/models/ir_ui_view.py", line 110, in render
    return super(View, self).render(values, engine=engine)
  File "/opt/odoo/addons/web_editor/models/ir_ui_view.py", line 26, in render
    return super(IrUiView, self).render(values=values, engine=engine)
  File "/opt/odoo/odoo/addons/base/ir/ir_ui_view.py", line 1049, in render
    return self.env[engine].render(self.id, qcontext)
  File "/opt/odoo/odoo/addons/base/ir/ir_qweb/ir_qweb.py", line 53, in render
    return super(IrQWeb, self).render(id_or_xml_id, values=values, **context)
  File "/opt/odoo/odoo/addons/base/ir/ir_qweb/qweb.py", line 248, in render
    self.compile(template, options)(self, body.append, values or {})
  File "/opt/odoo/odoo/addons/base/ir/ir_qweb/qweb.py", line 322, in _compiled_fn
    raise QWebException("Error to render compiling AST", e, path, node and etree.tostring(node[0]), name)
QWebException: (u'The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: medical.insurance.company, Operation: read)', None)
Traceback (most recent call last):
  File "/opt/odoo/odoo/addons/base/ir/ir_qweb/qweb.py", line 315, in _compiled_fn
    return compiled(self, append, values, options, log)
  File "<template>", line 1, in template_website_portal_medical_insurance_us_insurances_1075
  File "<template>", line 3, in body_call_content_1074
  File "<template>", line 15, in foreach_1073
  File "/opt/odoo/odoo/addons/base/ir/ir_qweb/ir_qweb.py", line 261, in _get_field
    content = converter.record_to_html(record, field_name, field_options)
  File "/opt/odoo/odoo/addons/base/ir/ir_qweb/fields.py", line 100, in record_to_html
    value = record[field_name]
  File "/opt/odoo/odoo/models.py", line 5177, in __getitem__
    return self._fields[key].__get__(self, type(self))
  File "/opt/odoo/odoo/fields.py", line 871, in __get__
    value = record._cache[self]
  File "/opt/odoo/odoo/models.py", line 5526, in __getitem__
    return value.get() if isinstance(value, SpecialValue) else value
  File "/opt/odoo/odoo/fields.py", line 48, in get
    raise self.exception
AccessError: (u'The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: medical.insurance.company, Operation: read)', None)

Error to render compiling AST
AccessError: (u'The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: medical.insurance.company, Operation: read)', None)
Template: website_portal_medical_insurance_us.insurances
Path: /templates/t/t/t[2]/div/t/div/div/div[2]/div[1]/div
Node: <div class="h5" t-field="plan.insurance_company_id.name"/>
```

Reply:
Dang yeah travis is showing the same thing.